src/views/transactions/components/transaction-overview/transaction-overview.view.tsx

```diff
@@ -170,16 +170,19 @@ function TransactionOverview({
     </Container>
   );
 
-  const footer = (buttonLabel: string): JSX.Element =>
+  const footer = (buttonLabel: string, loaderText: string): JSX.Element =>
     isTransactionBeingApproved ? (
       <div className={classes.signingSpinnerWrapper}>
         <Spinner />
-        <p className={classes.signingText}>Sign in your connected wallet to confirm transaction</p>
+        <p className={classes.signingText}>{loaderText}</p>
       </div>
     ) : (
       <PrimaryButton label={buttonLabel} onClick={handleFormSubmit} disabled={isButtonDisabled} />
     );
 
+  const l1TxLoaderText = "Sign in your connected wallet to confirm transaction";
+  const l2TxLoaderText = "Processing transaction";
```

Comment:
The behavior for L2 txs is a bit different. Instead of displaying the spinner with a text, we just need to show nothing but the submit button disabled. The button is going to appear already disabled as you will see in L118 and L121 that we are already disabling it for L2 txs.